test/CMakeTests.cmake

```diff
@@ -370,6 +376,17 @@ foreach (h5_test ${H5_TESTS})
           WORKING_DIRECTORY ${HDF5_TEST_BINARY_DIR}/H5TEST
       )
     endif ()
+
+    # Disable ctest timeouts for specific tests, such as multi-threaded
+    # tests, so that they can be run for as long as desired when the
+    # TestExpress level is set to 0. These tests must still use some
+    # other method to control test execution time when the TestExpress
+    # level is set to a different value.
+    if (h5_test IN_LIST H5TEST_NOTIMEOUT_TESTS)
+      set_tests_properties (H5TEST-${h5_test} PROPERTIES
```

Comment:
Disables the `ctest` timeout for mt_id_test specifically and assumes that the test will perform its own timeout checking.